sql/parser/builtins.go

```diff
@@ -951,6 +969,42 @@ func round(x float64, n int64) (Datum, error) {
 	return DFloat(y), err
 }
 
+// Pick the greatest or least value from a tuple.
+func pickFromTuple(greatest bool, args DTuple) (Datum, error) {
+	pos := -1
+	// Pick the first non-NULL value.
+	for i, d := range args {
+		if _, ok := d.(dNull); !ok {
+			pos = i
+			break
+		}
+	}
+	if pos == -1 {
+		return DNull, nil
+	}
+	g := args[pos]
+	// Pick a greater or smaller value.
+	for _, d := range args[pos+1:] {
+		if _, ok := d.(dNull); ok {
+			continue
+		}
+		var eval Datum
+		var err error
+		if greatest {
+			eval, err = evalComparison(LT, g, d)
+		} else {
+			eval, err = evalComparison(LT, d, g)
```

Comment:
Can you initialize the comparison op outside of the loop depending on `greatest`. Something like:

  cmp := LT
  if !greatest {
    cmp = GT
  }

  for ... {
    eval, err := evalComparison(cmp, g, d)
   ...
  }